ruma-events-macros/src/event_enum.rs

```diff
@@ -180,10 +188,26 @@ fn to_event_path(name: &LitStr) -> TokenStream {
         .map(|s| s.chars().next().unwrap().to_uppercase().to_string() + &s[1..])
         .collect::<String>();
 
-    let content_str = Ident::new(&format!("{}Event", event), span);
     let path = path.iter().map(|s| Ident::new(s, span));
-    quote! {
-        ::ruma_events::#( #path )::*::#content_str
+
+    match struct_name.to_string().as_str() {
+        "MessageEvent" | "MessageEventStub" if event_str.contains("redaction") => {
+            let redaction = if struct_name == "MessageEvent" {
+                quote! { RedactionEvent }
+            } else {
+                quote! { RedactionEventStub }
+            };
+            quote! { ::ruma_events::room::redaction::#redaction }
+        }
+        "ToDeviceEvent" | "StateEventStub" | "StrippedStateEventStub" | "MessageEventStub" => {
+            let path = path.clone();
```

Comment:
This seems pointless.

Reply:
It is, hmm odd at one point I needed that or it complained about moving the value.